auth.test.js

```diff
@@ -0,0 +1,43 @@
+const jsc = require('jsverify');
+
+const auth = require('./auth'),
+      allCreds = require('./config').creds;
+      
+let token = false;
+
+describe('handler', () => {
+
+  test('should generate a token with a valid user', () => {
+    token = auth.handler(allCreds.kate);
+    expect(token).toBeTruthy();
+  });
+
+  // this is dumb and there's probably a better way not using property testing
+  test('should return false if the user is not valid', () => {
+    let failedCreds = jsc.forall("json", (creds) => {
+      return auth.handler(creds);
+    });
+
+    jsc.throws(failedCreds);
+  });
+});
+
+describe('status', () => {
+  test('should return true if signature.verify returns true', () => {
+    let header = { 'authorization': token };
+    expect(auth.status(header)).toEqual(token);
+  });
+
+  test('should return false if signature.verfiy returns false', () => {
+    let failedToken = jsc.forall("string", (fakeToken) => {
```

Comment:
I like this test. Generating "fake tokens" and checking that they always fail is a useful property to document and ensure.

A minor nitpick: this property is not quite accurate. There are strings that are valid. While it is practically impossible for the random string generator to actually produce a valid token, the property does not help the reader of the code better understand the problem that the function is supposed to solve.

I find it more clear to only generate explicitly bad tokens. It might not be worth the effort in this specific instance (particularly because it is already tested in the signature module).

Reply:
is there a way to set a regex so we don't have a chance of generating a good one? say, setting a length but explicitly removing the `=` that is at the end of the tokens?